### PR TITLE
Fixes for the match grammar

### PR DESCRIPTION
**Pull request by brauliobz**
- added MatchArms to make things clearer
- added attributes to both the match block and match arms
- removed pipes before the patterns, since they're experimental

### CODE REVIEW

**Comment by goodmanjonathan** [comment]
Pipes before the patterns are being stabilized in 1.25.

**Comment by brauliobz** [comment]
> Pipes before the patterns are being stabilized in 1.25.

Are they really? Where can I follow that? Here:

https://github.com/rust-lang/rust/milestone/44 ?

I would prefer to only consider things stable when the stable build is published.

**Comment by goodmanjonathan** [comment]
[https://github.com/rust-lang/rust/pull/47947](https://github.com/rust-lang/rust/pull/47947)
[https://github.com/rust-lang/rust/pull/48374](https://github.com/rust-lang/rust/pull/48374)

I added the pipes as a prerequisite for the stabilization, so please keep them.

**Comment by brauliobz** [comment]
> I added the pipes as a prerequisite for the stabilization, so please keep them.

Ok, that makes all sense :+1: .

File: src/expressions/match-expr.md [hunk 1]
```diff
@@ -2,13 +2,26 @@
...
-> &nbsp;&nbsp; &nbsp;&nbsp; `}`  
+> _MatchArms_ :  
+> &nbsp;&nbsp; ( _MatchArm_ `=>` 
+>                             ( [_BlockExpression_] `,`<sup>?</sup>
+>                             | [_Expression_] `,` ) 
+>                           )<sup>\*</sup>  
+> &nbsp;&nbsp; _MatchArm_ `=>` ( [_BlockExpression_] | [_Expression_] ) `,`<sup>?</sup>  
+>  
+> _MatchArm_ :  
+> &nbsp;&nbsp; [_OuterAttribute_]<sup>\*</sup> _MatchArmPatterns_ _MatchArmGuard_
```
Comment (Havvy) [inline src/expressions/match-expr.md]: Can you give an example of the OuterAttribute here?

Comment (brauliobz) [reply src/expressions/match-expr.md]: I can't think of a useful example...

Comment (brauliobz) [reply src/expressions/match-expr.md]: I tried something like this, but the attribute doesn't apply to the use of deprecated `USBType::A` and rustc warns about both things instead (unused attribute and use of deprecated item):

```
enum USBType {
    #[deprecated] A,
    B,
    C,
}

fn describe(ty: USBType) -> &'static str {
    use USBType::*;
    match ty {
        #[allow(deprecated)] A => "",
        B => "",
        C => ""
    }
}
```

Comment (Havvy) [reply src/expressions/match-expr.md]: So it's allowed by the grammar, but doesn't seem to have a use case? Weird, but as long as it's legal...

**Comment by Havvy** [comment]
💟 Thanks!

**Pull request merged**